### PR TITLE
[PWA-2326]: Refactor deprecated type_id on ProductInterface 

### PR DESCRIPTION
**Pull request by pedrochiossi**
## Description

This PR updates graphQL queries and related functionality that uses deprecated `type_id` field from `ProductInterface` to use `__typename` field instead. 

## Related Issue

Closes PWA-2326

## Verification Steps

1. Ensure PWA Studio no longer uses the deprecated ProductInterface.type_id field

#### Test scenario(s) for direct fix/feature

1. Verify that searching for all different supported product types functions correctly (simple, configurable, etc.)

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/Gallery/useAddToCartButton.js [hunk 1]
```diff
@@ -21,10 +21,10 @@ import operations from './addToCart.gql';
  *
  */
 const UNSUPPORTED_PRODUCT_TYPES = [
```
Comment (fooman) [inline packages/peregrine/lib/talons/Gallery/useAddToCartButton.js:23]: Would be good to swap the logic around here as product types can be custom per store.

Comment (pedrochiossi) [reply packages/peregrine/lib/talons/Gallery/useAddToCartButton.js:23]: I agree @fooman, but I think it would be out of the scope of this task.

**Review by michaelyu0123 — APPROVED** [review]
LGTM

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**